src/desdemona/jobs/sample_submit_job.clj

```diff
@@ -1,33 +1,58 @@
+            [desdemona.tasks.kafka :refer [add-kafka-input add-kafka-output]]
+            [desdemona.tasks.core-async :refer [add-core-async-input add-core-async-output]]
+            [desdemona.tasks.sql :refer [add-sql-partition-input add-sql-insert-output]]
+            [desdemona.tasks.file-input :refer [add-seq-file-input]]
+            [desdemona.lifecycles.sample-lifecycle :refer [build-lifecycles]]
+            [desdemona.lifecycles.metrics :refer [add-metrics]]
+            [desdemona.lifecycles.logging :refer [add-logging]]
+            [desdemona.workflows.sample-workflow :refer [build-workflow]]
+            [aero.core :refer [read-config]]
             [onyx.api]))
 
-(defn submit-job [dev-env]
-  (let [dev-cfg (-> "dev-peer-config.edn" resource slurp read-string)
-        peer-config (assoc dev-cfg :onyx/id (:onyx-id dev-env))
-        ;; Turn :read-lines and :write-lines into core.async I/O channels
-        stubs [:read-lines :write-lines]
-        ;; Stubs the catalog entries for core.async I/O
-        dev-catalog (u/in-memory-catalog (build-catalog 20 50) stubs)
-        ;; Stubs the lifecycles for core.async I/O
-        dev-lifecycles (u/in-memory-lifecycles (build-lifecycles) dev-catalog stubs)]
-    ;; Automatically pipes the data structure into the channel, attaching :done at the end
-    (u/bind-inputs! dev-lifecycles {:read-lines dev-inputs/lines})
-    (let [job {:workflow workflow
-               :catalog dev-catalog
-               :lifecycles dev-lifecycles
-               :flow-conditions sf/flow-conditions
-               :task-scheduler :onyx.task-scheduler/balanced}]
-      (onyx.api/submit-job peer-config job)
-      ;; Automatically grab output from the stubbed core.async channels,
-      ;; returning a vector of the results with data structures representing
-      ;; the output.
-      (u/collect-outputs! dev-lifecycles [:write-lines]))))
+;;;;
+;; Lets build a job
+;; Depending on the mode, the job is built up in a different way
+;; When :dev mode, onyx-seq will be used as an input, with the meetup data being
+;; included in the onyx-seq lifecycle for easy access
+;; core.async is then added as an output task
+;;
+;; When using :prod mode, kafka is added as an input, and onyx-sql is used as the output
+
+(defn build-job [mode]
+  (let [batch-size 1
+        batch-timeout 1000
+        base-job {:catalog (build-catalog batch-size batch-timeout)
+                  :lifecycles (build-lifecycles {:mode mode})
+                  :workflow (build-workflow {:mode mode})
+                  :task-scheduler :onyx.task-scheduler/balanced}]
+    (cond-> base-job
+      (= :dev mode) (add-core-async-output :write-lines {:onyx/batch-size batch-size})
+      (= :dev mode) (add-seq-file-input :read-lines {:onyx/batch-size batch-size
+                                                     :filename "resources/sample_input.edn"})
+      (= :prod mode) (add-kafka-input :read-lines {:onyx/batch-size batch-size
```

Comment:
The README says "the tests automatically start and stop the development environment", which is probably the only place dev mode will be used.

We could rip it out and come up with a different way to run that test?